### PR TITLE
Add basic input actions and an XRI-based example scene

### PR DESCRIPTION
**Pull request by MattGrayUL**
## Summary

Adds a simple set of InputActions for using XRI, includes:

- Palm position
- Palm direction
- Is Pinching
- Is Grabbing
- Index finger tip position

Also includes and example scene to show the integration, uses indirect UI, indirect object and direct object interactions

## Contributor Tasks

_These tasks are for the merge request creator to tick off when creating a merge request._

- [ ] Pair review with a member of the QA team.
- [x] Add any release testing considerations to the MR for the next release.
- [x] Check any relevant CHANGELOG files have been updated.
- [x] Ensure documentation requirements are met e.g., public API is commented.
- [x] Consider any licensing/other legal implications for this MR e.g. notices required by any new libraries.
- [x] Add any relevant labels such as `breaking` to this MR.
- [ ] If this MR closes a Jira issue, make sure the fix version on the JIRA issue is set to the correct one.

## Reviewer Tasks

_Add any instructions or tasks for the reviewer such as specific test considerations before this can be merged._

[Use emojis in review threads to communicate intent and help contributors.](https://github.com/ultraleap/UnityPlugin/blob/develop/CONTRIBUTING.md#review-threads)

- [x] Code reviewed.
- [x] Non-code assets e.g. Unity assets/scenes reviewed.
- [ ] Documentation has been reviewed. Includes checking documentation requirements are met and not missing e.g., public API is commented.
- [ ] Checked and agree with release testing considerations added to MR for the next release.

## Closes JIRA Issue

Closes UNITY-695
Closes UNITY-966

### CODE REVIEW

File: Packages/Tracking Preview/Input Action Hands/Leap Hand Inputs.inputactions [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+                    "action": "Tracked",
+                    "isComposite": false,
+                    "isPartOfComposite": false
+                }
+            ]
+        }
+    ],
+    "controlSchemes": [
+        {
+            "name": "New control scheme",
```
Comment (MaxPalmer-UH) [inline Packages/Tracking Preview/Input Action Hands/Leap Hand Inputs.inputactions:263]: Should this be called "New control scheme"?

File: Packages/Tracking Preview/Input Action Hands/LeapHandUpdater.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+using UnityEngine;
+
+using Leap.Unity;
+using Leap;
+using UnityEngine.InputSystem;
+using UnityEngine.InputSystem.Layouts;
+
+namespace Leap.Unity.Preview.InputActions
+{
+    public class LeapHandUpdater : PostProcessProvider
```
Comment (MaxPalmer-UH) [inline Packages/Tracking Preview/Input Action Hands/LeapHandUpdater.cs:20]: Can we have comments on suitable public methods/classes please?

File: Packages/Tracking Preview/Input Action Hands/LeapHandUpdater.cs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                        _leftReportedNull = false;
+                        ConvertLeapHandToState(ref _leftState, hand);
+                        InputSystem.QueueStateEvent(LeapHandInput.leftHand, _leftState);
+                    }
+                    else
+                    {
+                        if (!_leftReportedNull)
+                        {
+                            _leftState.tracked = 0;
+                            InputSystem.QueueStateEvent(LeapHandInput.leftHand, _leftState);
```
Comment (MaxPalmer-UH) [inline Packages/Tracking Preview/Input Action Hands/LeapHandUpdater.cs:46]: Do we need to clear any other state info other than the tracked state when the hand is null?

File: Packages/Tracking Preview/Input Action Hands/Leap Hand Inputs.inputactions (hunk 1 above)
Comment (MattGrayUL) [reply Packages/Tracking Preview/Input Action Hands/Leap Hand Inputs.inputactions:263]: This file is auto-generated, but i'm happy to manually change the name :)

File: Packages/Tracking Preview/Input Action Hands/LeapHandUpdater.cs (hunk 3 above)
Comment (MattGrayUL) [reply Packages/Tracking Preview/Input Action Hands/LeapHandUpdater.cs:46]: I don't know whether we should/shouldn't essentially, 'tracked' should be being used to determine what data is up-to-date, so if it's false, you still may want the most recent data?

File: Packages/Tracking Preview/Input Action Hands/LeapHandInput.cs [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+    #endregion
+}
+    /// <summary>
+    /// An Input Device generated using two individual hand Input Devices representing left and right hands.
+    /// </summary>
+#if UNITY_EDITOR
+    [InitializeOnLoad]
+#endif
+    [InputControlLayout(displayName = "Ultraleap", stateType = typeof(LeapHandState), commonUsages = new[] { "LeftHand", "RightHand" }, canRunInBackground = true, isNoisy = true)]
```
Comment (MaxPalmer-UH) [inline Packages/Tracking Preview/Input Action Hands/LeapHandInput.cs:138]: The isNoisy property requires us to specify a minimum version for the com.unity.inputsystem on the asmdef - I think at least 1.4 (possibly 1.4.4). It's missing from 1.3.0 which is what I see when using this code.

Comment (MaxPalmer-UH) [reply Packages/Tracking Preview/Input Action Hands/LeapHandInput.cs:138]: https://docs.unity3d.com/Packages/com.unity.inputsystem@1.4/api/UnityEngine.InputSystem.Layouts.InputControlLayoutAttribute.html

File: Packages/Tracking Preview/XR Interaction Toolkit Integration/Editor/LeapInputActionUpdaterEditor.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+
+            inputLeapProviderProp = serializedObject.FindProperty("_inputLeapProvider");
+
+            directionSourceProp = serializedObject.FindProperty("directionSource");
+            isPinchingIsBinaryProp = serializedObject.FindProperty("isPinchingIsBinary");
+            isGrabbingIsBinaryProp = serializedObject.FindProperty("isGrabbingIsBinary");
```
Comment (BryceNewell-UL) [inline Packages/Tracking Preview/XR Interaction Toolkit Integration/Editor/LeapInputActionUpdaterEditor.cs:33]: test

**Review by BryceNewell-UL — APPROVED** [review]
These changes work, I am able to pinch but not grab the UI to spawn a sphere, I am then able to pull the sphere towards me with a grab then swap the sphere between hands with a direct interaction. Tested with a pico neo 3 running in editor. XRI scene causes "DrawMeshInstanced" errors in unity 2021.3.15f but this is expected.

**Comment by MaxPalmer-UH** [comment]
Example scene works but is incomplete - needs polish - e.g. text added to the scene to explain what's happening. Level that sphere falls onto seems weird as it sits infront of the indirect UI, etc.

**Comment by MattGrayUL** [comment]
@MaxPalmer-UH I have addressed the UI-focused points here, including the explanation text

The floor area is consistent with all other example scenes as it is part of the example assets prefab used elsewhere

I would say that for a Preview example, this is suitable as it is now, if we were to be putting this into the main Package we may also want to add some styling to the provided UI, but this will not be the case

**Pull request merged**